src/lsstseries/analysis/light_curve.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional
```

Comment:
Comment on the high level placement of this file (not this specific line): Would this make sense to go in top-level src directory as one of the fundamental data types?

Reply:
I can add a docstring for this class to explain the motivation for putting it here. The goal is for this to be a container class for use in analysis functions. So I _think_ that keeping it here makes sense. But if there are strong opinions, I'm happy to adjust.

Reply:
I don't feel strongly about this. So feel free to keep it where it is

Reply:
Gotcha, for now, kept where it is.